src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,100 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private float threshold;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (LOAD_FACTOR * table.length);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshold) {
+            resize();
+        }
+        Node<K, V> newNode = new Node(key, value, null);
+        int index = setHash(key);
+        if (table[index] == null) {
+            table[index] = newNode;
+            size++;
+            return;
+        }
+        Node<K, V> currentNode = table[index];
+        while (currentNode != null) {
+            if (isKeysIdentical(currentNode.key, key)) {
+                currentNode.value = value;
+                return;
+            }
+            if (currentNode.nextNode == null) {
+                currentNode.nextNode = newNode;
+                size++;
+            }
+            currentNode = currentNode.nextNode;
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int index = setHash(key);
+        Node<K, V> newNode = table[index];
+
+        if (index < 0 || index > table.length) {
+            throw new IndexOutOfBoundsException("Index " + index + " out of bounds");
+        }
+        while (newNode != null) {
+            if (isKeysIdentical(newNode.key, key)) {
+                return newNode.value;
+            }
+            newNode = newNode.nextNode;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        size = 0;
+        threshold *= (int) (LOAD_FACTOR * table.length);
+        Node<K, V>[] oldNode = table;
```

Comment:
Maybe would be better to rename the variable to **oldTable**, because it's not a node, it's the array of nodes